Vst modules

Paul-Licameli opened the pull request:
Resolves: #1131
Resolves: #1119
Resolves: #1118
Resolves: #1117
Resolves: #1116
Resolves: #1115
Resolves: #1098
Resolves: #1114
Resolves: #3874

Update:  combined with the Effect-modules branch.  That other PR is closed.  Rebased onto recent master, and conflicts resolved.  Build system changes, formerly part of this, are no longer needed.

Commits exclusive of the last three have already been reviewed and approved by @vsverchinsky and are unchanged besides conflict resolutions.

Seven modules for support of effect plug-in families and for the several built-in effects

The Audacity executable has five fewer dependencies on third-party libararies, as seen in this diagram:

<img width="283" alt="image" src="https://user-images.githubusercontent.com/11670369/205038375-757eb412-b479-4448-9c32-82f99e9a0667.png">

Two more effects modules extracted and four more third-party libraries eliminated from Audacity's dependencies.

<img width="348" alt="image" src="https://user-images.githubusercontent.com/11670369/205039511-43a0b66c-9026-4c8b-bad1-03679a97b891.png">


<!-- Use "x" to fill the checkboxes below like [x] -->

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

Paul-Licameli commented:
After Equalization refactoring was merged, this PR required a rebasing to resolve conflicts in the Eq* files only.

vsverchinsky reviewed (approved):
Approved with assumption that compilation error will be fixed (is this PR dependent on Equalization PR?). Though one suggestion. Now, when all effect plugin types are managed as modules, having two ways to enable/disable them is confusing, suggest to remove `Enable Effects` section from `Effects` in preferences.

Paul-Licameli commented:
> Approved with assumption that compilation error will be fixed (is this PR dependent on Equalization PR?). Though one suggestion. Now, when all effect plugin types are managed as modules, having two ways to enable/disable them is confusing, suggest to remove `Enable Effects` section from `Effects` in preferences.

I have rebased onto recent master, resolved a few simple conflicts in #include lines and in the visibility of RealtimeEffectList, fixed the Windows link, and force-pushed again.

I don't think we will follow your suggestion:  instead, I think the extension of the Modules preference page should be hidden in non-alpha builds.

The matter of module extraction needs a little discussion with the rest of the team, and again, I move this card back to review-in-progress until I get @crsib to agree to it as well.

Thanks for the review.
